### PR TITLE
Add SLM Start/Stop/Status APIs

### PR DESCRIPTION
**Pull request by codebrain**

### CODE REVIEW

File: src/CodeGeneration/ApiGenerator/RestSpecification/XPack/slm.get_status.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (Mpdreamz) [inline src/CodeGeneration/ApiGenerator/RestSpecification/XPack/slm.get_status.json]: I'm sure planned but create a PR for this to elastsearch as well.

Comment (codebrain) [reply src/CodeGeneration/ApiGenerator/RestSpecification/XPack/slm.get_status.json]: Yup, was going to add to the client first then open one in ES

Comment (codebrain) [reply src/CodeGeneration/ApiGenerator/RestSpecification/XPack/slm.get_status.json]: Opened in https://github.com/elastic/elasticsearch/pull/49759

File: src/Tests/Tests/XPack/Slm/SlmApiTests.cs [hunk 2]
```diff
@@ -167,6 +188,20 @@ public SlmApiTests(XPackCluster cluster, EndpointUsage usage) : base(new Coordin
...
 		});
 
+		[I] public async Task GetSnapshotLifecycleStatusResponse() => await Assert<GetIlmStatusResponse>(GetSnapshotLifecycleStatusStep, (v, r) =>
+		{
+			r.IsValid.Should().BeTrue();
+			r.ApiCall.HttpStatusCode.Should().Be(200);
+			r.OperationMode.Should().Be(LifecycleOperationMode.Running);
+		});
+
+		[I] public async Task StopSnapshotLifecycleResponse() => await Assert<StopIlmResponse>(StopSnapshotLifecycleStep, (v, r) =>
```
Comment (russcam) [inline src/Tests/Tests/XPack/Slm/SlmApiTests.cs]: Should be `StopSlmResponse`?

File: src/Tests/Tests/XPack/Slm/SlmApiTests.cs [hunk 3]
```diff
@@ -167,6 +188,20 @@ public SlmApiTests(XPackCluster cluster, EndpointUsage usage) : base(new Coordin
 			r.Policies.Should().NotBeNull().And.HaveCount(4).And.ContainKey(v);
 		});
 
+		[I] public async Task GetSnapshotLifecycleStatusResponse() => await Assert<GetIlmStatusResponse>(GetSnapshotLifecycleStatusStep, (v, r) =>
```
Comment (russcam) [inline src/Tests/Tests/XPack/Slm/SlmApiTests.cs]: Should be `GetSlmStatusResponse`?

**Comment by Mpdreamz** [comment]
```
Tests.XPack.Slm.SlmApiTests.GetSnapshotLifecycleStatusResponse
System.Exception : Nest.GetSlmStatusResponse is not expected response type Nest.GetIlmStatusResponse
```

```
Tests.XPack.Slm.SlmApiTests.StopSnapshotLifecycleResponse
System.Exception : Nest.StopSlmResponse is not expected response type Nest.StopIlmResponse
```

File: src/Tests/Tests/XPack/Slm/SlmApiTests.cs (hunk 3 above)
Comment (codebrain) [reply src/Tests/Tests/XPack/Slm/SlmApiTests.cs]: Fixed

File: src/Tests/Tests/XPack/Slm/SlmApiTests.cs (hunk 2 above)
Comment (codebrain) [reply src/Tests/Tests/XPack/Slm/SlmApiTests.cs]: Fixed

File: src/Elasticsearch.Net/Api/RequestParameters/RequestParameters.SnapshotLifecycleManagement.cs [hunk 4]
```diff
@@ -42,9 +42,27 @@ public class GetSnapshotLifecycleRequestParameters : RequestParameters<GetSnapsh
 		public override HttpMethod DefaultHttpMethod => HttpMethod.GET;
 	}
 
+	///<summary>Request options for GetStatus <para>https://www.elastic.co/guide/en/elasticsearch/reference/current/slm-get-status.html</para></summary>
+	public class GetSlmStatusRequestParameters : RequestParameters<GetSlmStatusRequestParameters>
```
Comment (russcam) [inline src/Elasticsearch.Net/Api/RequestParameters/RequestParameters.SnapshotLifecycleManagement.cs]: I think this should be `GetSnapshotLifecycleStatusRequestParameters`

File: src/Elasticsearch.Net/Api/RequestParameters/RequestParameters.SnapshotLifecycleManagement.cs [hunk 5]
```diff
@@ -42,9 +42,27 @@ public class GetSnapshotLifecycleRequestParameters : RequestParameters<GetSnapsh
...
+	}
+
 	///<summary>Request options for PutSnapshotLifecycle <para>https://www.elastic.co/guide/en/elasticsearch/reference/current/slm-api-put.html</para></summary>
 	public class PutSnapshotLifecycleRequestParameters : RequestParameters<PutSnapshotLifecycleRequestParameters>
 	{
 		public override HttpMethod DefaultHttpMethod => HttpMethod.PUT;
 	}
+
+	///<summary>Request options for Start <para>https://www.elastic.co/guide/en/elasticsearch/reference/current/slm-start.html</para></summary>
+	public class StartSlmRequestParameters : RequestParameters<StartSlmRequestParameters>
```
Comment (russcam) [inline src/Elasticsearch.Net/Api/RequestParameters/RequestParameters.SnapshotLifecycleManagement.cs]: I think this should be `StartSnapshotLifecycleRequestParameters`

File: src/Elasticsearch.Net/Api/RequestParameters/RequestParameters.SnapshotLifecycleManagement.cs [hunk 6]
```diff
@@ -42,9 +42,27 @@ public class GetSnapshotLifecycleRequestParameters : RequestParameters<GetSnapsh
...
+	{
+		public override HttpMethod DefaultHttpMethod => HttpMethod.POST;
+	}
+
+	///<summary>Request options for Stop <para>https://www.elastic.co/guide/en/elasticsearch/reference/current/slm-stop.html</para></summary>
+	public class StopSlmRequestParameters : RequestParameters<StopSlmRequestParameters>
```
Comment (russcam) [inline src/Elasticsearch.Net/Api/RequestParameters/RequestParameters.SnapshotLifecycleManagement.cs]: I think this should be `StopSnapshotLifecycleRequestParameters`

File: src/Nest/Descriptors.SnapshotLifecycleManagement.cs [hunk 7]
```diff
@@ -94,6 +94,14 @@ public GetSnapshotLifecycleDescriptor(): base()
 	// Request parameters
 	}
 
+	///<summary>Descriptor for GetStatus <para>https://www.elastic.co/guide/en/elasticsearch/reference/current/slm-get-status.html</para></summary>
+	public partial class GetSlmStatusDescriptor : RequestDescriptorBase<GetSlmStatusDescriptor, GetSlmStatusRequestParameters, IGetSlmStatusRequest>, IGetSlmStatusRequest
```
Comment (russcam) [inline src/Nest/Descriptors.SnapshotLifecycleManagement.cs]: `GetSnapshotLifecycleStatusDescriptor`

File: src/Nest/Descriptors.SnapshotLifecycleManagement.cs [hunk 8]
```diff
@@ -114,4 +122,20 @@ protected PutSnapshotLifecycleDescriptor(): base()
 		Id IPutSnapshotLifecycleRequest.PolicyId => Self.RouteValues.Get<Id>("policy_id");
 	// Request parameters
 	}
+
+	///<summary>Descriptor for Start <para>https://www.elastic.co/guide/en/elasticsearch/reference/current/slm-start.html</para></summary>
+	public partial class StartSlmDescriptor : RequestDescriptorBase<StartSlmDescriptor, StartSlmRequestParameters, IStartSlmRequest>, IStartSlmRequest
```
Comment (russcam) [inline src/Nest/Descriptors.SnapshotLifecycleManagement.cs]: `StartSnapshotLifecycleDescriptor`

File: src/Nest/Descriptors.SnapshotLifecycleManagement.cs [hunk 9]
```diff
@@ -114,4 +122,20 @@ protected PutSnapshotLifecycleDescriptor(): base()
...
+	{
+		internal override ApiUrls ApiUrls => ApiUrlsLookups.SnapshotLifecycleManagementStart;
+	// values part of the url path
+	// Request parameters
+	}
+
+	///<summary>Descriptor for Stop <para>https://www.elastic.co/guide/en/elasticsearch/reference/current/slm-stop.html</para></summary>
+	public partial class StopSlmDescriptor : RequestDescriptorBase<StopSlmDescriptor, StopSlmRequestParameters, IStopSlmRequest>, IStopSlmRequest
```
Comment (russcam) [inline src/Nest/Descriptors.SnapshotLifecycleManagement.cs]: `StopSnapshotLifecycleDescriptor`

File: src/Nest/Requests.SnapshotLifecycleManagement.cs [hunk 10]
```diff
@@ -129,6 +129,20 @@ public GetSnapshotLifecycleRequest(): base()
 	// Request parameters
 	}
 
+	[InterfaceDataContract]
+	public partial interface IGetSlmStatusRequest : IRequest<GetSlmStatusRequestParameters>
```
Comment (russcam) [inline src/Nest/Requests.SnapshotLifecycleManagement.cs]: `IGetSnapshotLifecycleStatusRequest`

File: src/Nest/Requests.SnapshotLifecycleManagement.cs [hunk 11]
```diff
@@ -129,6 +129,20 @@ public GetSnapshotLifecycleRequest(): base()
...
+	{
+	}
+
+	///<summary>Request for GetStatus <para>https://www.elastic.co/guide/en/elasticsearch/reference/current/slm-get-status.html</para></summary>
+	public partial class GetSlmStatusRequest : PlainRequestBase<GetSlmStatusRequestParameters>, IGetSlmStatusRequest
```
Comment (russcam) [inline src/Nest/Requests.SnapshotLifecycleManagement.cs]: `GetSnapshotLifecycleStatusRequest`

File: src/Nest/Requests.SnapshotLifecycleManagement.cs [hunk 12]
```diff
@@ -161,4 +175,32 @@ protected PutSnapshotLifecycleRequest(): base()
 		Id IPutSnapshotLifecycleRequest.PolicyId => Self.RouteValues.Get<Id>("policy_id");
 	// Request parameters
 	}
+
+	[InterfaceDataContract]
+	public partial interface IStartSlmRequest : IRequest<StartSlmRequestParameters>
```
Comment (russcam) [inline src/Nest/Requests.SnapshotLifecycleManagement.cs]: `IStartSnapshotLifecycleRequest`

File: src/Nest/Requests.SnapshotLifecycleManagement.cs [hunk 13]
```diff
@@ -161,4 +175,32 @@ protected PutSnapshotLifecycleRequest(): base()
...
+	{
+	}
+
+	///<summary>Request for Start <para>https://www.elastic.co/guide/en/elasticsearch/reference/current/slm-start.html</para></summary>
+	public partial class StartSlmRequest : PlainRequestBase<StartSlmRequestParameters>, IStartSlmRequest
```
Comment (russcam) [inline src/Nest/Requests.SnapshotLifecycleManagement.cs]: `StartSnapshotLifecycleRequest`

File: src/Nest/Requests.SnapshotLifecycleManagement.cs [hunk 14]
```diff
@@ -161,4 +175,32 @@ protected PutSnapshotLifecycleRequest(): base()
...
+	{
+		protected IStartSlmRequest Self => this;
+		internal override ApiUrls ApiUrls => ApiUrlsLookups.SnapshotLifecycleManagementStart;
+	// values part of the url path
+	// Request parameters
+	}
+
+	[InterfaceDataContract]
+	public partial interface IStopSlmRequest : IRequest<StopSlmRequestParameters>
```
Comment (russcam) [inline src/Nest/Requests.SnapshotLifecycleManagement.cs]: `IStopSnapshotLifecycleRequest`

File: src/Nest/Requests.SnapshotLifecycleManagement.cs [hunk 15]
```diff
@@ -161,4 +175,32 @@ protected PutSnapshotLifecycleRequest(): base()
...
+	{
+	}
+
+	///<summary>Request for Stop <para>https://www.elastic.co/guide/en/elasticsearch/reference/current/slm-stop.html</para></summary>
+	public partial class StopSlmRequest : PlainRequestBase<StopSlmRequestParameters>, IStopSlmRequest
```
Comment (russcam) [inline src/Nest/Requests.SnapshotLifecycleManagement.cs]: `StopSnapshotLifecycleRequest`

**Comment by codebrain** [comment]
Ported to `master`

**Pull request merged**